### PR TITLE
Feat: added more methods

### PR DESCRIPTION
**Pull request by ignaciosantise**
## Summary
* Added eth_signTypedData, eth_signTransaction
* Added modal showing the request status and response

### iOS

https://user-images.githubusercontent.com/25931366/235765561-ec19ee9a-30e6-4a71-8fb0-ff5b7e9327cc.MP4

### Android

https://user-images.githubusercontent.com/25931366/235766872-5a26e3ff-3039-46e3-acf6-d8538226c15b.mp4

### CODE REVIEW

File: example/src/types/methods.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+export interface FormattedRpcResponse {
+  method: string;
+  address: string;
+  valid: boolean;
+  result: string;
+  error?: string;
+}
+
+export interface FormattedRcpError {
```
Comment (crypblizz8) [inline example/src/types/methods.ts:11]:
```suggestion
export interface FormattedRpcError {
```

File: example/src/components/BlockchainActions.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
+import { useWeb3Modal } from '@web3modal/react-native';
+import { ethers } from 'ethers';
+import { useMemo, useState } from 'react';
+import { StyleSheet, Text, TouchableOpacity } from 'react-native';
+
+import type {
+  AccountAction,
+  FormattedRcpError,
```
Comment (crypblizz8) [inline example/src/components/BlockchainActions.tsx:8]: As above

File: example/src/components/BlockchainActions.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+export function BlockchainActions() {
+  const [rcpResponse, setRcpResponse] = useState<FormattedRpcResponse>();
+  const { provider } = useWeb3Modal();
+
+  const web3Provider = useMemo(
+    () => (provider ? new ethers.providers.Web3Provider(provider) : undefined),
+    [provider]
+  );
+
+  const [rcpError, setRcpError] = useState<FormattedRcpError>();
```
Comment (crypblizz8) [inline example/src/components/BlockchainActions.tsx:29]:
```suggestion
  const [rpcError, setRpcError] = useState<FormattedRpcError>();
```

File: example/src/components/BlockchainActions.tsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+  testEthSign,
+  testSendTransaction,
+  testSignMessage,
+  testSignTransaction,
+  testSignTypedData,
+} from '../utils/MethodUtil';
+import { RequestModal } from './RequestModal';
+
+export function BlockchainActions() {
+  const [rcpResponse, setRcpResponse] = useState<FormattedRpcResponse>();
```
Comment (crypblizz8) [inline example/src/components/BlockchainActions.tsx:21]:
```suggestion
  const [rpcResponse, setRpcResponse] = useState<FormattedRpcResponse>();
```

File: example/src/components/BlockchainActions.tsx [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+  );
+
+  const [rcpError, setRcpError] = useState<FormattedRcpError>();
+  const [loading, setLoading] = useState(false);
+  const [modalVisible, setModalVisible] = useState(false);
+
+  const onModalClose = () => {
+    setModalVisible(false);
+    setLoading(false);
+    setRcpResponse(undefined);
```
Comment (crypblizz8) [inline example/src/components/BlockchainActions.tsx:36]: This line and below too

File: example/src/components/BlockchainActions.tsx [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+      (
+        method: string,
+        rpcRequest: (
+          web3Provider: ethers.providers.Web3Provider
+        ) => Promise<FormattedRpcResponse>
+      ) =>
+      async () => {
+        if (!web3Provider) return;
+
+        setRcpResponse(undefined);
```
Comment (crypblizz8) [inline example/src/components/BlockchainActions.tsx:51]: .

File: example/src/components/BlockchainActions.tsx [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+      },
+      {
+        method: 'eth_signTypedData',
+        callback: wrapRpcRequest('eth_signTypedData', testSignTypedData),
+      },
+    ];
+  };
+
+  return (
+    <>
```
Comment (crypblizz8) [inline example/src/components/BlockchainActions.tsx:93]: Nitpick: Flatlist?

File: example/src/components/RequestModal.tsx [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+import Modal from 'react-native-modal';
+import type { FormattedRcpError, FormattedRpcResponse } from '../types/methods';
+import Close from '../assets/Close.png';
+
+interface Props {
+  rcpResponse?: FormattedRpcResponse;
+  isVisible: boolean;
+  onClose: () => void;
+  isLoading?: boolean;
+  rcpError?: FormattedRcpError;
```
Comment (crypblizz8) [inline example/src/components/RequestModal.tsx:18]: .

File: example/src/components/RequestModal.tsx [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+export function RequestModal({
+  rcpResponse,
```
Comment (crypblizz8) [inline example/src/components/RequestModal.tsx:22]: .

File: example/src/constants/eip712.ts [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+  },
+  primaryType: 'Mail',
+  domain: {
+    name: 'Ether Mail',
+    version: '1',
+    chainId: 1,
+    verifyingContract: '0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC',
+  },
+  message: {
+    from: { name: 'Cow', wallet: '0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826' },
```
Comment (crypblizz8) [inline example/src/constants/eip712.ts:28]: Is this a certain address?
Also nitpick but can we change to Alice so we stick to the Alice/Bob referencing

File: example/src/utils/MethodUtil.ts [hunk 11]
```diff
@@ -37,7 +55,77 @@ export const testSignMessage = async (
...
+  const [address] = await web3Provider.listAccounts();
+
+  if (!address) {
+    throw new Error('No address found');
+  }
+
+  const signature = await web3Provider.send('eth_sign', [address, hexMsg]);
+  const valid = verifyEip155MessageSignature(msg, signature, address);
+  return {
+    method: 'eth_sign (standard)',
```
Comment (crypblizz8) [inline example/src/utils/MethodUtil.ts:75]: Is this method string correct?

**Review by crypblizz8 — CHANGES_REQUESTED** [review]
LGTM, some comments + spelling regarding RPC. Great stuff!

File: example/src/constants/eip712.ts (hunk 10 above)
Comment (ignaciosantise) [reply example/src/constants/eip712.ts:28]: The example is taken from https://eips.ethereum.org/EIPS/eip-712, addresses too.

I can change to Alice 👍

**Pull request merged**